app/src/js/reducers/collections.js

```diff
@@ -1,156 +1,106 @@
 'use strict';
-import { set, del } from 'object-path';
-import assignDate from './assign-date';
 
+import { createReducer } from '@reduxjs/toolkit';
+import { deconstructCollectionId } from '../utils/format';
+import assignDate from './utils/assign-date';
 import {
-  COLLECTION,
-  COLLECTION_INFLIGHT,
-  COLLECTION_ERROR,
-  COLLECTION_APPLYWORKFLOW,
-  COLLECTION_APPLYWORKFLOW_INFLIGHT,
+  createClearItemReducer,
+  createErrorReducer,
+  createInflightReducer,
+  createSuccessReducer
+} from './utils/reducer-creators';
+import {
+  CLEAR_COLLECTIONS_FILTER,
+  CLEAR_COLLECTIONS_SEARCH,
   COLLECTION_APPLYWORKFLOW_ERROR,
-  COLLECTIONS,
-  COLLECTIONS_INFLIGHT,
+  COLLECTION_APPLYWORKFLOW_INFLIGHT,
+  COLLECTION_APPLYWORKFLOW,
+  COLLECTION_DELETE_ERROR,
+  COLLECTION_DELETE_INFLIGHT,
+  COLLECTION_DELETE,
+  COLLECTION_ERROR,
+  COLLECTION_INFLIGHT,
+  COLLECTION,
   COLLECTIONS_ERROR,
-  NEW_COLLECTION,
-  NEW_COLLECTION_INFLIGHT,
+  COLLECTIONS_INFLIGHT,
+  COLLECTIONS,
+  FILTER_COLLECTIONS,
   NEW_COLLECTION_ERROR,
-  COLLECTION_DELETE,
-  COLLECTION_DELETE_INFLIGHT,
-  COLLECTION_DELETE_ERROR,
-  UPDATE_COLLECTION,
-  UPDATE_COLLECTION_INFLIGHT,
-  UPDATE_COLLECTION_ERROR,
-  UPDATE_COLLECTION_CLEAR,
+  NEW_COLLECTION_INFLIGHT,
+  NEW_COLLECTION,
   SEARCH_COLLECTIONS,
-  CLEAR_COLLECTIONS_SEARCH,
-  FILTER_COLLECTIONS,
-  CLEAR_COLLECTIONS_FILTER
+  UPDATE_COLLECTION_CLEAR,
+  UPDATE_COLLECTION_ERROR,
+  UPDATE_COLLECTION_INFLIGHT,
+  UPDATE_COLLECTION,
 } from '../actions/types';
-import { createReducer } from '@reduxjs/toolkit';
-import { deconstructCollectionId } from '../utils/format';
 
 export const initialState = {
   list: {
     data: [],
     meta: {},
-    params: {}
+    params: {},
   },
   created: {},
   deleted: {},
   executed: {},
   map: {},
-  updated: {}
+  updated: {},
 };
 
 export default createReducer(initialState, {
-  [COLLECTION]: (state, action) => {
-    const { id, data } = action;
+  [COLLECTION]: ({ deleted, map }, { id, data }) => {
     const { name } = deconstructCollectionId(id);
     const collection = data.results.find((element) => element.name === name);
-    set(state, ['map', id, 'inflight'], false);
-    set(state, ['map', id, 'data'], assignDate(collection));
-    del(state, ['deleted', id]);
-  },
-  [COLLECTION_INFLIGHT]: (state, action) => {
-    const { id } = action;
-    set(state, ['map', id, 'inflight'], true);
-  },
-  [COLLECTION_ERROR]: (state, action) => {
-    const { id } = action;
-    set(state, ['map', id, 'inflight'], false);
-    set(state, ['map', id, 'error'], action.error);
-  },
-
-  [COLLECTION_APPLYWORKFLOW]: (state, action) => {
-    const { id } = action;
-    set(state, ['executed', id, 'status'], 'success');
-    set(state, ['executed', id, 'error'], null);
```

Comment:
so I started to look at the createCreateStatusReducer function, and I don't see how it sets the state.executed.id.error here.

I don't think this is working as you expect.
I added this test to the develop branch `test/reducers/collection.js`
It doesn't pass on this branch.

```
test('reducers/collections COLLECTION_APPLYWORKFLOW', (t) => {
  const testState = { ...initialState };
  const action = {
    type: 'COLLECTION_APPLYWORKFLOW',
    id: 'someid'
  };

  const expected = { ...initialState };
  expected.executed.someid = { status: 'success', error: null };

  const actual = reducer(testState, action);

  t.deepEqual(expected, actual);
});
```

Reply:
I pushed it up on a PR.
https://github.com/nasa/cumulus-dashboard/pull/738
You should make sure it can pass on this branch as well.

Reply:
I'll go take a look, but I believe the test fails because that nested object becomes
```
{ status: 'success' }
```
instead of
```
{ status: 'success', error: null }
```
There's no need to include `error: null`.

Reply:
I restored this.